### PR TITLE
Story/11162 http caching part1/11163 validation onboarded countries

### PR DESCRIPTION
**Pull request by pascal-brause**
## Description
Ports the call of onboardedCountries for the ticket validation from our old client to the new restService:
- Extend/Create `ValidationOnboardedCountriesResource`
- Create `ValidationOnboardedCountriesModel` which encapsulates the countries but also the CBOR decoding
- Remove in `HealthCertificateValidationOnboardedCountriesProvider` the hole success and failure handling, it is now done in the model or resource
- Move, adopt existing tests and add new tests to `ValidationOnboardedCountriesResource`
- Move `HealthCertificateValidationOnboardedCountriesError` to `ValidationOnboardedCountriesResource` and rename the error to `ValidationOnboardedCountriesError`
- Remove the caching layer object
- Remove the caches in the store
- Removes the old client call
- Removes the old request creation
- Removes the old url endpoint
- Added the new call to the developer menu for testing purposes

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11163

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/RuleValidation/CertLogic+Fakes.swift [hunk 1]
```diff
@@ -87,6 +87,16 @@ public var onboardedCountriesCBORDataFake: Data {
     return Data(cborCountries.encode())
 }
 
+public var onboardedCountriesCBORDataFake2: Data {
```
Comment (marcussc) [inline src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/RuleValidation/CertLogic+Fakes.swift:90]:
```suggestion
public var onboardedCountriesCBORDataFake_IT_UK: Data {
```

**Review by marcussc — APPROVED** [review]
Just a small naming improvement

File: src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/RuleValidation/CertLogic+Fakes.swift (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/RuleValidation/CertLogic+Fakes.swift:90]: good idea

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMNewHttpCommunication/DMNHCViewModel.swift [hunk 2]
```diff
@@ -57,18 +57,18 @@ final class DMNHCViewModel {
 					}
 				}
 			)
-		case .otpEdusAuthorization:
+		case .validationOnboardedCountries:
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/DMNewHttpCommunication/DMNHCViewModel.swift:60]:
```suggestion
		case .validateOnboardedCountries:
```

**Pull request merged**